monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -895,6 +917,30 @@ object Task extends TaskInstances {
   def gatherUnordered[A](in: TraversableOnce[Task[A]]): Task[List[A]] =
     TaskGatherUnordered(in)
 
+  /** Given a `TraversableOnce[A]` and a function `A => Task[B]`,
+    * nondeterministically apply the function to each element of the collection
+    * without keeping the original ordering of the results.
+    *
+    * This function is similar to [[wander]], but neither the effects nor the
+    * results will be ordered. Useful when you don't need ordering because:
+    *
+    *  - it has non-blocking behavior (but not wait-free)
+    *  - it can be more efficient (compared with [[wander]]), but not
+    *    necessarily (if you care about performance, then test)
+    *
+    * It's a generalized version of [[gatherUnordered]].
+    */
+  def wanderUnordered[A, B, M[X] <: TraversableOnce[X]](in: M[A])(f: A => Task[B])
+    (implicit cbf: CanBuildFrom[M[A], B, M[B]]): Task[M[B]] = {
```

Comment:
I would not go with a version with `CanBuildFrom` and simply follow the normal `gatherUnordered` signature.

This is new in master and the version 2.1 I'm going to release (probably today), but the reason is that we are no longer using `CanBuildFrom` in `gatherUnordered` and returning a `Task[List[A]]` instead is because it better reflects what the function does, because the result really is a `List`, being kept in an `Atomic` and updated in a non-blocking fashion and if the user wants something else, he can convert that `List` himself, with whatever cost that incurs and that cost will be visible, whereas if you hide it in this function, that cost will not be visible.

So basically traversing that `List` to build the collection you want is not worth it imho. It's also the defining characteristic versus normal `gather`. I mean originally `gather` was built on top of `gatherUnordered`, following Scalaz in this regard, but now `gather` has a pretty efficient version too, so the defining characteristics of `gatherUnordered` are:

1. non-blocking
2. more memory efficient, since we traverse the original collection only once and build the final `List` as we go along, only once

Let me know your thoughts on this matter.